### DISCUSSION TITLE
[May 2024] zkSync Developer update

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#568] [Announcements]
Hello developers! Welcome to the May edition of the zkSync Developer Update, your once-a-month post to stay up to date with the latest info about protocol updates, a range of new and improved development tools, recent integrations, and a variety of tutorials.

Here are the updates for the last month:

### ⚙️ Protocol upgrade v24 is live on mainnet

The latest protocol upgrade now available on mainnet (after a few delays), with a bunch of new features like:

- Precompiles `ecParing` and `p256Verify` ([RIP7212](https://github.com/ethereum/RIPs/blob/master/RIPS/rip-7212.md)) that enable hardware encryption from smartphones for wallets like Clave.
- BridgeHub, a new L1 smart contract bridge that will be used to enable interoperability between chains created with ZK Stack.
- Support for up to 6 blobs (EIP4844).

[Read the full details in this thread](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/519).

### 🧰 Developer tools

- **hardhat plugins**: we’ve released multiple updates to our Hardhat plugins:
    - `hardhat-zksync-deploy`: now supports passing the deployment type (`create`, `create2`, `createAccount` or `create2Account`).
    - `hardhat-zksync-node` : added the `—tag` flag to use a specific tag of the era-test-node.
    - `hardhat-zksync-upgradable` : extended to support different deployment types and `salt`
    
    Find [[more info in our docs](https://docs.zksync.io/build/tooling/hardhat/hardhat-zksync.html).
    
- **Viem**: further updates in `viem/zksync` to support more L2 and L1 public actions like `getBridgehubContractAddress` , `getDefaultBridgeAddresses` and `getL1Allowance`. More info in [the viem zkSync docs](https://viem.sh/zksync).
- **Go SDK SmartAccount class:** this class was released a few weeks ago in `zksync-ethers` and we’ve included it in our Go SDK. This class simplifies the usage of native account abstraction with pre-configured ECDSA and multisig smart contract accounts. [Learn more in our docs](https://docs.zksync.io/build/sdks/js/zksync-ethers/accounts.html#smartaccount).
- **Remix plugin update:** More updates from the Nethermind team on the zkSync Remix plugin, including automatic smart contract verification in the zkSync explorer and support for `payable` methods. And we have more updates coming soon  👀
- **Herodotus Turbo**: this tool enables smart contracts to access current and historical data located on other chains using storage proofs. Read the [full announcement here](https://herodotusdev.medium.com/turbo-transactions-launch-on-zksync-testnet-4f7694560366).

### 🌿 Lens network

Lens protocol is coming to the zkSync ecosystem with its own chain powered by ZK Stack. Read [the full announcement and technical details here](https://app.t2.world/article/clw6l2z0018727620mc1dq26xr2).

![CleanShot 2024-06-05 at 13 31 09@2x](https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/2de56cb7-826f-483c-8962-df6d8002dc02)

[Source](https://x.com/LensProtocol/status/1790458242855751950)

### 🛡️zkSync public node

We released the beta version of the zkSync public node so anyone can run a zkSync node to monitor network activity. The node is a read-replica that external parties run and can be used to have low latency access to the zkSync network’s data for testing purposes as well as building third-party apps. This is one of the first steps towards the network decentralisation. 

Find the system requirements and instructions [in our docs](https://docs.zksync.io/infra/quickstart.html).

### 👩‍💻 Akindo WaveHack

We’re sponsoring the Akindo WaveHack online hackathon with up to $50.000 in prizes. Submissions remain open as the hackathon will run through the end of August, with prizes being awarded in waves every two weeks. 

Join by building projects on the following bounty tracks:

- [Write Builder Tutorials or Technical Content](https://app.akindo.io/wave-hacks/o6dMDWpWOUq22G6v8)
- [Build an application using Native Account Abstraction](https://app.akindo.io/wave-hacks/3dXlN2mMqFVAwx63)
- [Build a Game or SocialFi application](https://app.akindo.io/wave-hacks/lr7Vx0NZWc0eBgDRz)

### ⚗️ Alchemy University videos

The Alchemy University team released two educational videos about native account abstraction on zkSync. 

- [Native paymasters on zkSync](https://www.youtube.com/watch?v=7rC9rjPtxBs): learn how to create a minimal paymaster that pays the transaction fees for all accounts.
- [Smart accounts on zkSync](https://www.youtube.com/watch?v=GbOs_tVJUvA): an in-depth tutorial on how to create a smart account with custom logic.

![CleanShot 2024-06-07 at 12 18 42@2x](https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/44ebc59c-243d-4925-822f-661b3f86f794)

Stay tuned for more videos from the Alchemy team.

### ⬡ Chainlink hackathon workshops

As part of the Chainlink BlockMagic hackathon, we gave two workshops for all the developers participating. Here are the recordings:

- [Developer Experience on zkSync](https://www.youtube.com/watch?v=YAou1sjErAY)
- [Exploring Data Feeds on zkSync](https://www.youtube.com/watch?v=oCqaPogmwCQ)

### 🔗 Stay connected

- Follow [[@zkSyncDevs](https://twitter.com/zkSyncDevs)](https://twitter.com/zkSyncDevs) on X
- Join the #dev-general channel in [[our Discord](https://join.zksync.dev/)](https://join.zksync.dev/)

Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.

Keep buidling! 👩‍💻🧑🏻‍💻

**Comment by LegalM0ney** [comment]
very good machine
